### PR TITLE
Adding generic_logs template for hints autodiscovery

### PR DESCRIPTION
**Pull request by gizas**
Adding Documentation for generic_logs template for Elastic Agent Hints Autodiscovery scenario in order to collect logs from k8s pods

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 1]
```diff
@@ -81,7 +81,7 @@ and avoid placing passwords in plain text.
 
 The stream to use for logs collection, for example, stdout/stderr.
 
-If the specified package has no logs support, a generic container's logs input will be used as a fallback.
+If the specified package has no logs support, a generic container's logs input will be used as a fallback. See example `Hints autodiscovery for kubernetes log collection` below.
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:84]:
```suggestion
If the specified package has no logs support, a generic container's logs input will be used as a fallback. See the `Hints autodiscovery for kubernetes log collection` example below.
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 2]
```diff
@@ -168,6 +171,143 @@ After deploying this Pod, the data will start flowing in automatically. You can
 NOTE: All assets (dashboards, ingest pipelines, and so on) related to the Redis integration are not installed. You need to explicitly <<install-uninstall-integration-assets,install them through {kib}>>.
 
 
+[discrete]
+=== Hints autodiscovery for kubernetes log collection
+
+The log collection for Kubernetes autodiscovered pods can be supported by using  https://github.com/elastic/elastic-agent/tree/main/deploy/kubernetes/elastic-agent-standalone/templates.d/container_logs.yml[container_logs.yml template]. Users need to emit a caontainer_logs mapping so as to start collecting logs for all the discovered containers *even if no annotations are present in the containers*.
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:177]:
```suggestion
The log collection for Kubernetes autodiscovered pods can be supported by using  https://github.com/elastic/elastic-agent/tree/main/deploy/kubernetes/elastic-agent-standalone/templates.d/container_logs.yml[container_logs.yml template]. Users need to emit a container_logs mapping so as to start collecting logs for all the discovered containers *even if no annotations are present in the containers*. 
```

Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:177]: Users do not emit any mappings right? This sentence does not look correct.

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 3]
```diff
@@ -168,6 +171,143 @@ After deploying this Pod, the data will start flowing in automatically. You can
...
+[source,yaml]
+----
+providers.kubernetes:
+  node: ${NODE_NAME}
+  scope: node
+  hints:
+    enabled: true
+----
+
+Example to manually mount the needed container_logs template under the inputs.d folder of Elastic Agent
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:189]: Why we need this extra mount? All templates are mounted automatically under `/etc/elastic-agent/inputs.d` if we uncomment the respective parts in the k8s manifests. So I see no need to specifically mount this one.

**Review by ChrsMark — CHANGES_REQUESTED** [review]

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc (hunk 3 above)
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:189]: I think that if we want an example with custom template's addition we should create a separate section for this and be specific there.

**Comment by ChrsMark** [comment]
Also shall we completely remove the respective part from the plain policy of standalone manifests? Like in https://github.com/elastic/elastic-agent/blob/8d538f7befb4859d7cd0aef68658010d2ac5e1c6/deploy/kubernetes/elastic-agent-kustomize/default/elastic-agent-standalone/base/elastic-agent-standalone-daemonset-configmap.yaml#L362-L383?

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc (hunk 2 above)
Comment (gizas) [reply docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc]: How about now?

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc (hunk 3 above)
Comment (gizas) [reply docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc]: Ok removed mount info and updated the section a bit

**Comment by gizas** [comment]
@ChrsMark nice idea: I have removed them in relevant PR: https://github.com/elastic/elastic-agent/pull/2986/files

**Pull request merged**